src/engines/TrainElectricEngine.cpp

```diff
@@ -24,42 +23,353 @@ namespace godot {
+
+        ClassDB::bind_method(
+                D_METHOD("set_transducer_input_voltage"), &TrainElectricEngine::set_transducer_input_voltage);
+        ClassDB::bind_method(
+                D_METHOD("get_transducer_input_voltage"), &TrainElectricEngine::get_transducer_input_voltage);
+        ADD_PROPERTY(
+                PropertyInfo(Variant::FLOAT, "power/transducer/input_voltage"), "set_transducer_input_voltage",
+                "get_transducer_input_voltage");
+
+        ClassDB::bind_method(D_METHOD("set_accumulator_recharge_source"), &TrainElectricEngine::set_accumulator_recharge_source);
+        ClassDB::bind_method(D_METHOD("get_accumulator_recharge_source"), &TrainElectricEngine::get_accumulator_recharge_source);
+        ADD_PROPERTY(
+                PropertyInfo(
+                        Variant::INT, "power/accumulator/recharge_source", PROPERTY_HINT_ENUM,
+                        "NotDefined,InternalSource,Transducer,Generator,Accumulator,CurrentCollector,PowerCable,Heater,"
+                        "Main"),
+                        "set_accumulator_recharge_source", "get_accumulator_recharge_source");
+
+        ClassDB::bind_method(D_METHOD("set_power_cable_power_source"), &TrainElectricEngine::set_power_cable_power_source);
+        ClassDB::bind_method(D_METHOD("get_power_cable_power_source"), &TrainElectricEngine::get_power_cable_power_source);
+        ADD_PROPERTY(
+                PropertyInfo(
+                        Variant::INT, "power/power_cable/source", PROPERTY_HINT_ENUM,
+                        "NoPower,BioPower,MechPower,ElectricPower,SteamPower"
+                        "Main"),
+                        "set_power_cable_power_source", "get_power_cable_power_source");
+
+        ClassDB::bind_method(
+                D_METHOD("set_power_cable_steam_pressure"), &TrainElectricEngine::set_power_cable_steam_pressure);
+        ClassDB::bind_method(
+                D_METHOD("get_power_cable_steam_pressure"), &TrainElectricEngine::get_power_cable_steam_pressure);
+        ADD_PROPERTY(
+                PropertyInfo(Variant::FLOAT, "power/power_cable/steam_pressure"), "set_power_cable_steam_pressure",
+                "get_power_cable_steam_pressure");
     }
 
     void TrainElectricEngine::_do_fetch_state_from_mover(TMoverParameters *mover, Dictionary &state) {
         TrainEngine::_do_fetch_state_from_mover(mover, state);
 
         state["compressor_enabled"] = mover->CompressorFlag;
         state["compressor_allowed"] = mover->CompressorAllow;
-
         state["converter_enabled"] = mover->ConverterFlag;
         state["converted_allowed"] = mover->ConverterAllow;
         state["converter_time_to_start"] = mover->ConverterStartDelayTimer;
+        state["power_source"] = static_cast<int>(mover->EnginePowerSource.SourceType);
+        state["accumulator/recharge_source"] = static_cast<int>(mover->EnginePowerSource.RAccumulator.RechargeSource);
```

Comment:
W `state` nie stosowałem do tej pory slashy. Sam podkreślnik (`_`) by tu wystarczył. 

Background: mam przyzwyczajenie z jezykow dynamicznych, że można się do takich proposów dostać jak do atrybutu, gdzie `/` nie będzie prawidłowym tokenem nazwy atrybutu. Do tego jak zserializujesz `state` do czegokolwiek, to alfabet nazw propsów będzie `[a-z0-9_]`, co raczej nie stworzy problemu w żadnym formacie. Slash może już wprowadzić kłopot.

Z drugiej strony `/` możne być użyty jako separator dla określenia group. Trzeba jednak zdecydować że powyższe uwagi nas/tego projektu nie dotyczą i że mamy stosować `/` - wtedy trzeba będzie zmienić implementacje pozostałych klas.